src/inverted-index.js

```diff
@@ -0,0 +1,140 @@
+    if (!utils.isValid(fileName, jsonFile)) {
+      return false;
+    }
+    if (typeof jsonFile === 'string') {
+      jsonFile = JSON.parse(jsonFile);
+    }
+    this.jsonDatabase[fileName] = [];
+    Object.keys(jsonFile).forEach((keys) => {
+      this.jsonDatabase[fileName].push(jsonFile[keys]);
+    });
+    return true;
+  }
+
+  /**
+   * [getJsonDatabase function to return the saved uploads]
+   * @return {object} [the saved uploads]
+   */
+  getJsonDatabase() {
+    return this.jsonDatabase;
+  }
+
+  /**
+   * [createIndex Creates an index of the words in the received json file]
+   * @param  {string}   filePath [the key(filename) of the json value to index]
+   * @param  {Function} cb  [call back to return the indexed file object/an html format index table]
+   * @return {array} [an arrray of the indexed file result and the html Div of the index]
+   */
+  createIndex(filePath) {
+    const indexFile = this.indexFile;
+    const jsonDoc = this.jsonDatabase[filePath];
+    let joinedValues = '';
+    let wordArray = [];
+    if (indexFile[filePath]) {
+      return false;
+    }
+    indexFile[filePath] = {};
+    jsonDoc.forEach((element, index) => {
+      joinedValues = utils.cleanString((`${element.title} ${element.text}`));
+      wordArray = new Set(joinedValues.split(' '));
+      wordArray.forEach((word) => {
+        indexFile[filePath][word] = indexFile[filePath][word] || [];
+        indexFile[filePath][word].push(index);
+      });
+    });
+    this.indexFile = indexFile;
+    return indexFile[filePath];
+  }
+
+  /**
+   * [getIndex Gets the index object of the indexed json file]
+   * @param  {string} fileName [the filename(key) of the index needed]
+   * @return {object}          [index of the object]
+   */
+  getIndex(fileName) {
+    return this.indexFile[fileName] || this.indexFile;
+  }
+
+  /**
+   * [searchIndex It searches the already indexed files for particular words]
+   * @param  {string}    fileNames     [description]
+   * @param  {Function}  cb            [description]
+   * @param  {...Array} searchContent [the words to search for]
+   * @return {Array}                  [an array of two elements, an
+   * object with the search term as key and their locations in the
+   * array of the originally uploaded file, an html view of the result]
+   */
+  searchIndex(fileNames, ...searchContent) {
+    const searchResult = {};
+    let searchTerms = searchContent.join(' ');
+    if (fileNames.length < 1) {
+      fileNames = this.getFilenames();
+    }
+    searchTerms = utils.cleanString(searchTerms, /[^a-z0-9\s,]+/gi);
+    searchTerms = searchTerms.split(/[,\s]/);
+    searchTerms.forEach((searchTerm) => {
+      searchResult[searchTerm] = {};
+      fileNames.forEach((fileName) => {
+        if (this.indexFile[fileName][searchTerm]) {
+          searchResult[searchTerm][fileName] = this.indexFile[fileName][searchTerm];
+        }
+      });
+    });
+    this.searchResult = searchResult;
```

Comment:
To further buttress my point about your not needing `this.searchResult`. You could just simply return `searchResult` here cos I don't see anywhere else where you need `searchResult`